### PR TITLE
ref(sampling): Remove sampling mode 'total'

### PR DESCRIPTION
**Pull request by jjbayer**
Stop supporting dynamic sampling mode `"total"`, which adjusted for the client sample rate.

* Old downstream Relays will default to `"received"` now that the field is missing (it was not serialized in this case anyway).
* Sentry has already stopped sending this field: https://github.com/getsentry/sentry/blob/1f2bc3e56e111e42b5fa1bb50ac5ae9f393f1a0b/src/sentry/relay/config/__init__.py#L346

### CODE REVIEW

File: relay-server/src/services/processor/dynamic_sampling.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ use crate::utils::{self, sample, ItemAction, SamplingResult};
 /// the main project state.
 ///
 /// If there is no transaction event in the envelope, this function will do nothing.
-pub fn normalize(state: &mut ProcessEnvelopeState<TransactionGroup>) {
+pub fn ensure_dsc(state: &mut ProcessEnvelopeState<TransactionGroup>) {
```
Comment (iker-barriocanal) [inline relay-server/src/services/processor/dynamic_sampling.rs:44]: 👍

**Review by iker-barriocanal — APPROVED** [review]
Thanks for the cleanup and fixing the changelog!

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
 ## Unreleased
 
-
 **Bug fixes:**
 
 - Respect country code TLDs when scrubbing span tags. ([#3458](https://github.com/getsentry/relay/pull/3458))
 
 **Features**:
 
+- **Breaking change:** Stop supporting dynamic sampling mode `"total"`, which adjusted for the client sample rate. ([#3474](https://github.com/getsentry/relay/pull/3474))
```
Comment (olksdr) [inline CHANGELOG.md]: Maybe it's better to add additional section for this? I think we did it in the past

File: relay-server/src/utils/dynamic_sampling.rs [hunk 3]
```diff
@@ -76,13 +76,8 @@ pub fn is_trace_fully_sampled(
...
     }
 
-    let adjustment_rate = match root_project_config.mode {
-        SamplingMode::Total => dsc.sample_rate,
-        _ => None,
-    };
-
     // TODO(tor): pass correct now timestamp
-    let evaluator = SamplingEvaluator::new(Utc::now()).adjust_client_sample_rate(adjustment_rate);
+    let evaluator = SamplingEvaluator::new(Utc::now());
```
Comment (olksdr) [inline relay-server/src/utils/dynamic_sampling.rs:81]: the TODO on top of this line, can it be removed? Or what does the TODO even mean?

**Review by olksdr — APPROVED** [review]

File: relay-server/src/utils/dynamic_sampling.rs (hunk 3 above)
Comment (jjbayer) [reply relay-server/src/utils/dynamic_sampling.rs:81]: I noticed it as well, I will look into it.

File: CHANGELOG.md (hunk 2 above)
Comment (jjbayer) [reply CHANGELOG.md]: I copied this from the 24.4.0 section, but I can also make a separate section if you prefer.

Comment (olksdr) [reply CHANGELOG.md]: In [`23.4.0`](https://github.com/getsentry/relay/blob/master/CHANGELOG.md#2340) it was separate section, and I find it a bit better and easier to spot for out users. What do you think?

Comment (jjbayer) [reply CHANGELOG.md]: Alright, added a new section and updated 24.4.0 section as well.

**Pull request merged**